### DISCUSSION TITLE
Update Companion Error

### DISCUSSION THREAD

**Discussion by winkelement** [requarks/wiki#3437] [Help / Questions]
**Describe the bug**
Trying to upgrade Wiki.js from 2.4.107 to 2.5.144 via the Update Companion docker container fails.

**To Reproduce**
1. Create and start the docker containers as described here https://docs.requarks.io/install/ubuntu with tag 2.4
2. Go to https://yourwiki.com/a/system
3. You will be offered the option to upgrade ("Wiki.js can perform the upgrade to the latest version for you."). Click "PERFORM UPGRADE".
4. Animation will start, no Error in the UI, page will reload after a while and the version will still be same. Looking at the docker status will reveal `wiki-update-companion` container has crashed.

**Expected behavior**
"PERFORM UPGRADE" should successfully perform an upgrade.

**Screenshots**
N/A

**Host Info (please complete the following information):**
 - OS: Docker
 - Wiki.js version: 2.4.107
 - Database engine: postgres 12.3

**Additional context**
Output of `docker logs wiki-update-companion` after crash

> info: Microsoft.Hosting.Lifetime[0]
>       Now listening on: http://[::]:80
> info: Microsoft.Hosting.Lifetime[0]
>       Application started. Press Ctrl+C to shut down.
> info: Microsoft.Hosting.Lifetime[0]
>       Hosting environment: Production
> info: Microsoft.Hosting.Lifetime[0]
>       Content root path: /app
> Unhandled exception. System.Net.Http.HttpRequestException: Connection failed
>  ---> System.Net.Internals.SocketExceptionFactory+ExtendedSocketException (111): Connection refused /var/run/docker.sock
>    at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
>    at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)
>    at System.Net.Sockets.Socket.UnsafeBeginConnect(EndPoint remoteEP, AsyncCallback callback, Object state, Boolean flowContext)
>    at System.Net.Sockets.Socket.BeginConnect(EndPoint remoteEP, AsyncCallback callback, Object state)
>    at System.Net.Sockets.Socket.ConnectAsync(EndPoint remoteEP)
>    at System.Net.Sockets.SocketTaskExtensions.ConnectAsync(Socket socket, EndPoint remoteEP)
>    at Docker.DotNet.DockerClient.<>c__DisplayClass6_0.<<-ctor>b__1>d.MoveNext()
> --- End of stack trace from previous location where exception was thrown ---
>    at Microsoft.Net.Http.Client.ManagedHandler.ProcessRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
> debug2: channel 0: window 999375 sent adjust 49201
>    --- End of inner exception stack trace ---
>    at Microsoft.Net.Http.Client.ManagedHandler.ProcessRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
>    at Microsoft.Net.Http.Client.ManagedHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
>    at System.Net.Http.HttpClient.FinishSendAsyncUnbuffered(Task`1 sendTask, HttpRequestMessage request, CancellationTokenSource cts, Boolean disposeCts)
>    at Docker.DotNet.DockerClient.PrivateMakeRequestAsync(TimeSpan timeout, HttpCompletionOption completionOption, HttpMethod method, String path, IQueryString queryString, IDictionary`2 headers, IRequestContent data, CancellationToken cancellationToken)
>    at Docker.DotNet.DockerClient.MakeRequestForStreamAsync(IEnumerable`1 errorHandlers, HttpMethod method, String path, IQueryString queryString, IRequestContent body, IDictionary`2 headers, TimeSpan timeout, CancellationToken token)
>    at Docker.DotNet.Models.StreamUtil.MonitorStreamForMessagesAsync[T](Task`1 streamTask, DockerClient client, CancellationToken cancel, IProgress`1 progress)
>    at wiki_update_companion.Controllers.UpgradeController.Post() in /src/Controllers/UpgradeController.cs:line 44
>    at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__139_1(Object state)
>    at System.Threading.QueueUserWorkItemCallback.<>c.<.cctor>b__6_0(QueueUserWorkItemCallback quwi)
>    at System.Threading.ExecutionContext.RunForThreadPoolUnsafe[TState](ExecutionContext executionContext, Action`1 callback, TState& state)
>    at System.Threading.QueueUserWorkItemCallback.Execute()
>    at System.Threading.ThreadPoolWorkQueue.Dispatch()
>    at System.Threading._ThreadPoolWaitCallback.PerformWaitCallback()
>

**Comment by NGPixel** [comment]
The companion app container is not able to connect to the Docker socket. Have you mounted the docker socket correctly? Is it readable by the container?
